### ISSUE TITLE
[web] CanvasKit Text Search broken on Chrome (Find... / Find on page)

### ISSUE THREAD

**Issue by creativecreatorormaybenot** [flutter/flutter#72274]
The `Find...` feature in Chrome is completely broken when using CanvasKit:

![image](https://user-images.githubusercontent.com/19204050/102091673-0f7dba80-3e17-11eb-8181-f63ae5d1e626.png)

## Problems

1. When just launching `Find...` on a CanvasKit Flutter web app and trying to search for text, it will not work **at all**. You have to _select text_ (using [`SelectableText`](https://api.flutter.dev/flutter/material/SelectableText-class.html)).
2. When you have selected text in a [`SelectableText`](https://api.flutter.dev/flutter/material/SelectableText-class.html), it will only work for **that** text widget (not find anything anywhere else).
3. When it works, it will fail to properly render the results: the results will be offset, i.e. not where the actual text is.
4. The marker might not be rendered at all when the widget clips.
5. Cannot find text in normal `Text` widgets at all.
6. (Addition for Firefox:) On Firefox, it is even worse because **any** scrolling will reset the search.

## Example

Here is an example I recorded. I think the problems should be obvious.

![](https://s8.gifyu.com/images/ezgif-1-d78397e20cbc.gif)

1. Occurs right at the beginning: even though there clearly is `Flutter` written in text on the screen, no results are found.
2. That is what happens throughout the recording: I select different `SelectableText`s and use `Find...`. It always only finds text in the one I last selected text in.
3. Especially visible on the top right card.
4. Again, top right card third match.
5. It says "Flutter" in other texts, but they are not selectable and so I will never be able to find it there.

The site I used for the example is [accessible here](https://creativecreatorormaybenot.github.io/portfolio/#/) (it just happens to be the only CanvasKit web app I have deployed).

### Doctor

```
 • Chrome (web)     • chrome     • web-javascript • Google Chrome 87.0.4280.88

[✓] Flutter (Channel beta, 1.24.0-10.2.pre, locale en-US)
    • Flutter version 1.24.0-10.2.pre
    • Framework revision 022b333a08 (4 weeks ago), 2020-11-18 11:35:09 -0800
    • Engine revision 07c1eed46b
    • Dart version 2.12.0 (build 2.12.0-29.10.beta)
````

**Comment by TahaTesser** [comment]
Hi @creativecreatorormaybenot 
Thanks for filing the issue, I can confirm al the  problems

<details> 
<summary>code sample</summary> 

```dart
import 'package:flutter/material.dart';

void main() => runApp(MyApp());

class MyApp extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      title: 'Material App',
      theme: ThemeData.dark(),
      home: Home(),
    );
  }
}

class Home extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(
        title: Text('Material App Bar'),
      ),
      body: Center(
        child: Container(
          child: SelectableText(
              'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Suspendisse facilisis auctor ipsum non convallis. Fusce ut dolor in lacus blandit rutrum sit amet sit amet ipsum. Sed risus eros, euismod eu suscipit quis, tempus vitae dui. Maecenas accumsan tempus neque eu venenatis. Nunc mauris elit, accumsan quis elit at, finibus gravida nulla. Suspendisse massa nisi, ultricies gravida dignissim id, tempor quis urna. Aenean efficitur malesuada lorem eget dictum. Proin quis magna in sem finibus condimentum vel quis enim. Morbi et eros nisl. Mauris mattis nunc eget nibh vehicula mollis. Curabitur cursus mi ligula. Etiam efficitur volutpat justo, sit amet pharetra quam mollis sed. Donec egestas sollicitudin arcu, nec condimentum sapien cursus ac.'),
        ),
      ),
      floatingActionButton: FloatingActionButton(
        child: Icon(Icons.add),
        onPressed: () {},
      ),
    );
  }
}

``` 

</details>



<details> 
<summary>multiple flutter doctor -v</summary> 

```bash
[✓] Flutter (Channel dev, 1.25.0-8.0.pre, on macOS 11.0.1 20B29 darwin-x64,
    locale en-GB)
    • Flutter version 1.25.0-8.0.pre at /Users/tahatesser/Code/flutter_dev
    • Framework revision a12e2a473a (6 days ago), 2020-12-08 23:13:04 -0500
    • Engine revision df39e5c515
    • Dart version 2.12.0 (build 2.12.0-133.0.dev)

[✓] Android toolchain - develop for Android devices (Android SDK version 30.0.2)
    • Android SDK at /Users/tahatesser/Code/sdk
    • Platform android-30, build-tools 30.0.2
    • ANDROID_HOME = /Users/tahatesser/Code/sdk
    • Java binary at: /Applications/Android
      Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build
      1.8.0_242-release-1644-b3-6915495)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 12.2)
    • Xcode at /Volumes/Extreme/Xcode.app/Contents/Developer
    • Xcode 12.2, Build version 12B45b
    • CocoaPods version 1.10.0

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 4.1)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build
      1.8.0_242-release-1644-b3-6915495)

[!] Android Studio
    • Android Studio at /Applications/Android/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    ✗ Android Studio not found at /Applications/Android/Contents
    • Try updating or re-installing Android Studio.

[✓] VS Code (version 1.52.0)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.17.0

[✓] Connected device (3 available)
    • Taha’s iPad (mobile) • 00008020-000255113EE8402E • ios            • iOS
      14.2
    • macOS (desktop)      • macos                     • darwin-x64     • macOS
      11.0.1 20B29 darwin-x64
    • Chrome (web)         • chrome                    • web-javascript • Google
      Chrome 87.0.4280.88

! Doctor found issues in 1 category.
```

```bash
[✓] Flutter (Channel master, 1.25.0-9.0.pre.43, on macOS 11.0.1 20B29
    darwin-x64, locale en-GB)
    • Flutter version 1.25.0-9.0.pre.43 at /Users/tahatesser/Code/flutter_master
    • Framework revision 63062a6443 (24 hours ago), 2020-12-13 23:19:13 +0800
    • Engine revision 4797b06652
    • Dart version 2.12.0 (build 2.12.0-141.0.dev)

[✓] Android toolchain - develop for Android devices (Android SDK version 30.0.2)
    • Android SDK at /Users/tahatesser/Code/sdk
    • Platform android-30, build-tools 30.0.2
    • ANDROID_HOME = /Users/tahatesser/Code/sdk
    • Java binary at: /Applications/Android
      Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build
      1.8.0_242-release-1644-b3-6915495)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 12.2)
    • Xcode at /Volumes/Extreme/Xcode.app/Contents/Developer
    • Xcode 12.2, Build version 12B45b
    • CocoaPods version 1.10.0

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 4.1)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build
      1.8.0_242-release-1644-b3-6915495)

[!] Android Studio
    • Android Studio at /Applications/Android/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    ✗ Android Studio not found at /Applications/Android/Contents
    • Try updating or re-installing Android Studio.

[✓] VS Code (version 1.52.0)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.17.0

[✓] Connected device (3 available)
    • Taha’s iPad (mobile) • 00008020-000255113EE8402E • ios            • iOS
      14.2
    • macOS (desktop)      • macos                     • darwin-x64     • macOS
      11.0.1 20B29 darwin-x64
    • Chrome (web)         • chrome                    • web-javascript • Google
      Chrome 87.0.4280.88

! Doctor found issues in 1 category.
```

</details>

**Comment by yjbanov** [comment]
Deduplicating with https://github.com/flutter/flutter/issues/65504

**Comment by creativecreatorormaybenot** [comment]
@yjbanov Will CanvasKit be considered specially, though? Afaic there are completely different problems between CanvasKit and DOM rendering, not sure, however.
